src/services/bundler.ts

```diff
@@ -1,61 +1,184 @@
-import JSZip from "jszip";
+import JSZip, { JSZipObject } from "jszip";
+import toml from "toml";
 
-const MAX_FILE_SIZE = 0x10000000; // 256MB
+import { convertFiles, isFontFile, isImageFile } from "./utilities";
 
-export function validateZip(file: File): Promise<string | void> {
-  const jszip = new JSZip();
-  return new Promise((resolve, reject) => {
-    if (file.size > MAX_FILE_SIZE) {
-      reject("Your archive is too big!");
-      return;
-    }
-    jszip
-      .loadAsync(file)
-      .then((zip: JSZip) => {
-        if (zip.files["lovebrew.toml"] === undefined) {
-          reject("Missing configuration file");
-        } else {
-          resolve();
-        }
-      })
-      .catch(() => {
-        resolve("Invalid archive. Refer to the documentation.");
-      });
-  });
+export async function validateZip(file: File) {
+  const zip = await JSZip.loadAsync(file);
+
+  if (zip.files["lovebrew.toml"] === undefined) {
+    return Promise.reject("Missing configuration file.");
+  }
+
+  return Promise.resolve();
```

Comment:
No need to `return Promise.reject()` or `Promise.resolve()` inside an `async` function. Use `throw Error()` instead of rejecting and don't return anything if you don't need to return a value - `async` wraps it all in a `Promise` under the hood.